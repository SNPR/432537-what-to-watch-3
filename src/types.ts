export type Movie = {
  name: string;
  posterUrl: string;
  previewUrl: string;
  bigPosterUrl: string;
  backgroundColor: string;
  description: string;
  rating: number;
  votes: number;
  director: string;
  starring: string[];
  runTime: string;
  genre: string;
  releaseYear: number;
  id: number | string;
  isFavorite: boolean;
  videoUrl: string;
  trailerUrl: string;
};
