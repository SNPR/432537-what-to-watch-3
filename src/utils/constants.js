export const ALL_GENRES = `All genres`;

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  ADD_REVIEW: `/add-review`,
  MY_LIST: `/mylist`,
  FILMS: `/films`,
  PLAYER: `/player`
};

export const SHOWED_MOVIES_DEFAULT = 8;

export const MIN_REVIEW_LENGTH = 50;
export const MAX_REVIEW_LENGTH = 400;

export const Tab = {
  Name: {
    OVERVIEW: `overview`,
    DETAILS: `details`,
    REVIEWS: `reviews`
  },
  ACTIVE_CLASS: `movie-nav__item--active`
};
