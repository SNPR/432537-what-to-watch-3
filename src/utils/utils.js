export const getTextRating = (value) => {
  if (value >= 0 && value < 3) {
    return `Bad`;
  } else if (value >= 3 && value < 5) {
    return `Normal`;
  } else if (value >= 5 && value < 8) {
    return `Good`;
  } else if (value >= 8 && value <= 10) {
    return `Very good`;
  }

  return ``;
};
