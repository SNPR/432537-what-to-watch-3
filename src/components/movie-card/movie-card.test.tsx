import * as React from "react";
import * as renderer from "react-test-renderer";
import MovieCard from "./movie-card";
import {Movie} from "../../types";
import {noop} from "../../utils/utils";

const movie: Movie = {
  name: `MovieName`,
  posterUrl: `https://url.com`,
  previewUrl: `https://url.com`,
  bigPosterUrl: `https://url.com`,
  backgroundColor: `blue`,
  description: `Descrtiption`,
  rating: 9,
  votes: 3452,
  director: `Director`,
  starring: [`Artist 1`, `Artist 2`],
  runTime: `2h 30m`,
  genre: `Action`,
  releaseYear: 1995,
  id: 1,
  isFavorite: false,
  videoUrl: `https://url.com`,
  trailerUrl: `https://url.com`
};

it(`Should render MovieCard component`, () => {
  const tree = renderer
    .create(
        <MovieCard
          movie={movie}
          onMovieCardClick={noop}
          onMovieCardMouseOver={noop}
          onMovieCardMouseOut={noop}
          isPlaying={true}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
