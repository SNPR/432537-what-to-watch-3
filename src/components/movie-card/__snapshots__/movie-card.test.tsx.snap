// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MovieCard component 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      autoPlay={true}
      controls={true}
      muted={true}
      onClick={[Function]}
      poster="https://url.com"
      width="100%"
    >
      <source
        src="https://url.com"
      />
    </video>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="#"
    >
      MovieName
    </a>
  </h3>
</article>
`;
